src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1202,6 +1209,95 @@ DatabaseShardImp::findAcquireIndex(
     return boost::none;
 }
 
+void
+DatabaseShardImp::scheduleFinalizeShard(std::uint32_t shardIndex)
+{
+    taskQueue_->addTask([this, shardIndex]
+    {
+        auto retry = [this, shardIndex]
+        {
+            const uint8_t TIMER_INTERVAL = 60;
```

Comment:
Should probably make sure we aren't stopping here before setting the timer.

eg. 
`        if (isStopping())
            return;
`